### ISSUE TITLE
Modify the default Kubernetes cluster domain is `cluster.local`.

### ISSUE THREAD

**Issue by marandalucas** [mongodb/mongodb-kubernetes-operator#1207]
Hi, I hope you are doing well.

**Does anyone know if we can modify the "clusterDomain" as in MongoDB Enterprise Operator?**

https://github.com/mongodb/mongodb-enterprise-kubernetes/blob/2b06c4b3bc6b0ddba24fb03435b04f2e2175ebfc/samples/mongodb/persistent-volumes/replica-set-persistent-volumes.yaml#L16-L19

Apparently, it is hardcoded in the code into the MongoDB OpenSource Operator.
https://github.com/mongodb/mongodb-kubernetes-operator/search?q=clusterDomain

We would like to move from Enterprise Operator to OpenSource Operator, but this is one of our requirements in our environment, and unfortunately, the replicaset does not work properly.

Thanks in advance!

**Comment by irajdeep** [comment]
Hi @marandalucas, it is possible to configure the cluster domain in the community operator as well; however, in community it's possible to change it via an environment variable `CLUSTER_DOMAIN`. unline CR spec in the enterprise. Please check the release notes of operator v0.7.1 for more details: https://github.com/mongodb/mongodb-kubernetes-operator/releases/tag/v0.7.1

**Comment by irajdeep** [comment]
Feel free to re-open if the issue still persists.

**Comment by marandalucas** [comment]
Thank you so much. Now, it's working!!!

I miss in the documentation something about this configuration. For me, it's not clear at all :( 
**Example in our MongoDB Operator:**

```
# Downloaded from https://raw.githubusercontent.com/mongodb/mongodb-kubernetes-operator/v0.7.8/config/manager/manager.yaml
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    email: support@mongodb.com
  labels:
    owner: mongodb
  name: mongodb-kubernetes-operator
  namespace: operators
spec:
  replicas: 1
  selector:
    matchLabels:
      name: mongodb-kubernetes-operator
  strategy:
    rollingUpdate:
      maxUnavailable: 1
    type: RollingUpdate
  template:
    metadata:
      labels:
        name: mongodb-kubernetes-operator
    spec:
      affinity:
        podAntiAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
          - labelSelector:
              matchExpressions:
              - key: name
                operator: In
                values:
                - mongodb-kubernetes-operator
            topologyKey: kubernetes.io/hostname
      containers:
      - command:
        - /usr/local/bin/entrypoint
        env:
        - name: CLUSTER_DOMAIN
          value: "appspace-cloud-private-na2-a.appspace.cluster.local"
        - name: WATCH_NAMESPACE
          value: "*"
        - name: POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        - name: OPERATOR_NAME
          value: mongodb-kubernetes-operator
        - name: AGENT_IMAGE
          value: quay.io/mongodb/mongodb-agent:12.0.15.7646-1
        - name: VERSION_UPGRADE_HOOK_IMAGE
          value: quay.io/mongodb/mongodb-kubernetes-operator-version-upgrade-post-start-hook:1.0.6
        - name: READINESS_PROBE_IMAGE
          value: quay.io/mongodb/mongodb-kubernetes-readinessprobe:1.0.12
        - name: MONGODB_IMAGE
          value: mongo
        - name: MONGODB_REPO_URL
          value: docker.io
        image: quay.io/mongodb/mongodb-kubernetes-operator:0.7.8
        imagePullPolicy: Always
        name: mongodb-kubernetes-operator
        resources:
          limits:
            cpu: 1100m
            memory: 1Gi
          requests:
            cpu: 500m
            memory: 200Mi
        securityContext:
          readOnlyRootFilesystem: true
          runAsUser: 2000
          allowPrivilegeEscalation: false
      securityContext:
        seccompProfile:
          type: RuntimeDefault
      serviceAccountName: mongodb-kubernetes-operator
```

**Comment by dan-mckean** [comment]
Hi, I'm Dan and I'm the Product Manager for MongoDB's support of Kubernetes.

I'm doing some work right now to try and identify how the Community Operator is being used. The Community Operator is something I inherited when I started at MongoDB, but it doesn't get as much attention from us as we'd like and we're trying to understand how it's used in order to establish its future. It will help us prioritize future issues and PRs raised by the community 🙂

Here's a super short survey (it's much easier for us to review all the feedback that way!): [https://docs.google.com/forms/d/e/1FAIpQLSfwrwyxBSlUyJ6AmC-eYlgW_3JEdfA48SB2i5--_WpiynMW2w/viewform?usp=sf_link](https://docs.google.com/forms/d/e/1FAIpQLSfwrwyxBSlUyJ6AmC-eYlgW_3JEdfA48SB2i5--_WpiynMW2w/viewform?usp=sf_link)

 If you'd rather email me instead: [dan.mckean@mongodb.com](mailto:dan.mckean@mongodb.com?subject=MongoDB%20Community%20Operator%20feedback)

Thank you in advance!
Dan
